lego_loam_ip:
  ros__parameters:
    # leishen16
    laser:
        odom_type: "laser_odometry"         #wheel_odometry/laser_odometry 
        baselink_frame: "base_link"
        num_vertical_scans: 16
        num_horizontal_scans: 1000
        ground_scan_index: 7
        vertical_angle_bottom: -15.0          # degrees
        vertical_angle_top: 15.0            # degrees
        scan_period: 0.1                   # seconds

    imageProjection:
        segment_valid_point_num: 5
        segment_valid_line_num: 2
        segment_theta: 60.0                      # decrese this value may improve accuracy
        maximum_detection_range: 120.0            # point cloud within this range will be used to project image
        distance_for_patch_between_rings: 1.0    # patch ground between rings within this range, i.e. ring0 and ring1 is smaller than 1 meter, patch them

lego_loam_fa:
  ros__parameters:
  
    featureAssociation:
        edge_threshold: 0.1
        surf_threshold: 0.1
        nearest_feature_search_distance: 3.0
        
    mapping:
        to_map_optimization: true

lego_loam_mo:
  ros__parameters:
    mapping:

        distance_between_key_frame: 1.0
        angle_between_key_frame: 1.0

        enable_loop_closure: true

        surrounding_keyframe_search_num: 10        
        history_keyframe_search_radius: 15.0  
        history_keyframe_search_num: 5
        history_keyframe_fitness_score: 0.5
        ground_voxel_size: 0.4
        ground_edge_threshold_num: 60